### DISCUSSION TITLE
this is a questies

### DISCUSSION THREAD

**Discussion by zaze06** [cc-tweaked/CC-Tweaked#774] [Q&A] [answered]
sens I can't make an account on the computer craft website and I'm banned to make questions on StackOverflow I'm asking to hear

How can I get the first item in the inventory that isn't on a list?

and yes I will try to fix the problem whit StackOverflow

**Comment by Lupus590** [comment]
This would have been better as a [discussion](https://github.com/SquidDev-CC/CC-Tweaked/discussions) instead of an issue.

There are new forums: https://forums.computercraft.cc/index.php

Discord is also a thing: https://discord.computercraft.cc/

Also, please work on your spelling, I can understand that it can be difficult if English is not your first language. You might find [Grammarly](https://app.grammarly.com/) helpful.

**Reply by zaze06** [reply]
yes I'm using gramma rly but it didn't find any issues with the text. thx for a link to discord I tried to find one but I didn't find it.

**Reply by Lupus590** [reply]
Are you sure that you have Grammarly set up correctly? I'm fairly sure that it should be flagging a few things in what you've posted.

**Comment by SquidDev** [comment]
Hrmr, what do you mean by "isn't on a list" here? Unless you're talking about [Russel's Paradox](https://en.wikipedia.org/wiki/Russell%27s_paradox), I'm not quite sure what you mean.

A good starting point is probably the documentation on inventories, especially the [`list` method](https://tweaked.cc/generic_peripheral/inventory.html#v:list)

**Comment by fatboychummy** [comment, marked as answer]
I'm assuming you're looking at how to check if an item in a chest or etc is not in a blacklist?

If so, initialize a table of all the item identifiers which are "bad" (this may change between versions), then you will want to loop through the inventory and the items to check if:

* On each slot, is there an item that is blacklisted?

Something like this might work (you *will* need to modify it). My code assumes 1.12.2 with Plethora (as I've not had too much of a chance to mess with 1.16), so if you're on newer versions of mc, some of the code may need to be different.

```lua
local blacklist = {
  ["minecraft:dirt:0"] = true, -- these strings are in the format "modid:blockid:damage"
  ["minecraft:stone:0"] = true,
  ... -- whichever other items
}

local chest = peripheral.find("chest")
local list = chest.list()

for i = 1, chest.size() do -- for each slot in the chest
  local item = list[i]
  if item then -- if there is an item in this slot
    if not blacklist[string.format("%s:%s", item.name, item.metadata)] then -- if the item is NOT in the blacklist
      -- do something with the item.
    end
  end
end
```

**Reply by zaze06** [reply]
Thx this was what I ment and I could have used the word "not in a black list" insted of "not in a list"
